JS/Lesson18/esra/script.js

```diff
@@ -35,7 +44,7 @@ function concatenateStrings(array) {
   for (let i = 0; i < array.length; i++) {
     sentence = sentence + ' ' + array[i];
   }
-  return sentence; 
+  return sentence.replaceAll(' !', '!'); 
```

Comment:
Good try with removing unnecessary spaces before ! character.
In reality, there would be more characters that you need to check. E.g. '.', ',', ':', '?', etc.
You could, for example, create an array of these characters and then check if the array includes the current symbol:
```
const specialCharacters = ['.', ',', ':', '?', '!', ';'];
.... 
const isSpecialCharacter = specialCharacters.includes(array[i]);
if (isSpecialCharacter) {
sentence = sentence + array[i];
} else {
sentence = sentence + ' ' + array[i];
}
```

You don't need to change the code now, it's more a suggestion for the future!